arex-instrumentation/httpclient/arex-httpclient-ning/src/main/java/io/arex/inst/httpclient/ning/ResponseFutureListener.java

```diff
@@ -0,0 +1,28 @@
+package io.arex.inst.httpclient.ning;
+
+import com.ning.http.client.ListenableFuture;
+import io.arex.agent.bootstrap.ctx.TraceTransmitter;
+import io.arex.inst.runtime.log.LogManager;
+
+import java.util.concurrent.*;
+
+public class ResponseFutureListener implements Runnable {
+    private final ListenableFuture<?> responseFuture;
+    private final NingHttpClientExtractor extractor;
+    private final TraceTransmitter transmitter;
+    public ResponseFutureListener(NingHttpClientExtractor extractor, ListenableFuture<?> responseFuture) {
+        this.responseFuture = responseFuture;
+        this.extractor = extractor;
+        this.transmitter = TraceTransmitter.create();
+    }
+
+    @Override
+    public void run() {
+        try (TraceTransmitter transmit = transmitter.transmit()){
+            extractor.record(responseFuture.get(1, TimeUnit.SECONDS));
+        } catch (Exception e) {
+            LogManager.warn("ResponseFutureListener", e);
+            Thread.currentThread().interrupt();
```

Comment:
not recommended to interrupt directly, unless catched is `InterruptedException` exception, otherwise may interrupt normal business logic, 
`responseFuture.addListener(new ResponseFutureListener(extractor, responseFuture), DirectExecutor.INSTANCE)`
because `DirectExecutor.INSTANCE` not using thread pool, still executed directly by the current thread